model_analyzer/result/results.py

```diff
@@ -0,0 +1,167 @@
+# Copyright (c) 2022 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from model_analyzer.constants import LOGGER_NAME
+
+import logging
+
+logger = logging.getLogger(LOGGER_NAME)
+
+
+class Results:
+    """
+    Provides storage and accessor functions for measurements
+    """
+    MODEL_CONFIG_INDEX = 0
+    MEASUREMENTS_INDEX = 1
+
+    def __init__(self):
+        """          
+        """
+        self._results = {}
+        self._done = {}
+
+    @classmethod
+    def from_dict(cls, results_dict):
+        results = Results()
+        results._results = results_dict['_results']
+
+        return results
+
+    def add_measurement(self, run_config, key, measurement):
+        """
+        Given a RunConfig and a key store a measurement
+        
+        Parameters
+        ----------
+        run_config: RunConfig
+        key: str
+        measurement: Measurement
+        """
+        model_name, model_config, model_config_name = self._extract_run_config_fields(
+            run_config)
+
+        if model_name not in self._results:
+            self._results[model_name] = {}
+            self._done[model_name] = False
+
+        if model_config_name not in self._results[model_name]:
+            self._results[model_name][model_config_name] = (model_config, {})
+
+        self._results[model_name][model_config_name][
+            Results.MEASUREMENTS_INDEX][key] = measurement
+
+    def contains_model(self, model_name):
+        """
+        Checks if the model name exists
+        in the results
+
+        Parameters
+        ----------
+        model_name : str
+            The model name
+
+        Returns
+        -------
+        bool
+        """
+        return model_name in self._results
+
+    def contains_model_config(self, model_name, model_config_name):
+        """
+        Checks if the model name and model config name
+        exists in the results
+        
+        Parameters
+        ----------
+        model_name : str
+            The model name
+            
+        model_config_name: str
+            The model config name 
+            
+        Returns
+        -------
+        bool
+        """
+        return model_config_name in self._results[model_name]
+
+    def is_done(self, model_name):
```

Comment:
What is the value of turning this into something that behaves like a generator? Can't we just return the whole list and have the user naturally iterate over it?